web-local/src/lib/components/navigation/dashboards/MetricsDefinitionAssets.svelte

```diff
@@ -105,21 +109,18 @@
     );
   };
 
-  const deleteMetricsDef = (id: string) => {
+  const deleteMetricsDef = (metricsDef: MetricsDefinitionEntity) => {
     if (
       $applicationStore.activeEntity.type === EntityType.MetricsDefinition &&
-      $applicationStore.activeEntity.id === id
+      $applicationStore.activeEntity.id === metricsDef.id
     ) {
-      const nextMetricsDefId = getNextEntityId($metricsDefinitions, id);
-
-      if (nextMetricsDefId) {
-        goto(`/dashboard/${nextMetricsDefId}`);
-      } else {
-        goto("/");
-      }
+      goto(`/model/${metricsDef.sourceModelId}`);
```

Comment:
@AdityaHegde oddly this doesn't seem to work. Any clue what might be happening?

Reply:
You need to delete the code in metrics-definition-api that does somewhat similar thing or move this there.